src/components/file-viewer/file-viewer.tsx

```diff
@@ -0,0 +1,394 @@
+ *:::
+ * @exampleComponent limel-example-file-viewer
+ * @exampleComponent limel-example-file-viewer-office
+ * @exampleComponent limel-example-file-viewer-filename
+ * @exampleComponent limel-example-file-viewer-actions
+ * @exampleComponent limel-example-file-viewer-with-picker
+ */
+
+@Component({
+    tag: 'limel-file-viewer',
+    shadow: true,
+    styleUrl: 'file-viewer.scss',
+})
+export class FileViewer {
+    /**
+     * Link to the file
+     */
+    @Prop({ reflect: true })
+    public url: string;
+
+    /**
+     * The name of the file that also contains its extension.
+     * Overrides the filename provided by the URL.
+     * Useful when the URL does not contain the filename.
+     */
+    @Prop({ reflect: true })
+    public filename: string;
+
+    /**
+     * Alternative text for assistive technologies and screen readers,
+     * used for image files.
+     */
+    @Prop({ reflect: true })
+    public alt: string;
+
+    /**
+     * Defines the localization for translations.
+     */
+    @Prop()
+    public language: Languages = 'en';
+
+    /**
+     * Defines the third-party viewer that should be used to render
+     * the content of office files, such as word processing documents,
+     * presentations, or spreadsheets.
+     */
+    @Prop()
+    public officeViewer: OfficeViewer = 'microsoft-office';
+
+    /**
+     * An array of custom actions that can be displayed
+     * as an action menu on the file which is being displayed.
+     */
+    @Prop()
+    public actions: ListItem[];
+
+    /**
+     *
+     */
+    @Prop()
+    public webserverDomain: string;
+
+    /**
+     * determines if the document is hosted on the same domain as the current webserver
+     */
+    @State()
+    public sameDomain: boolean = false;
+
+    /**
+     *
+     */
+    @Event()
+    public action: EventEmitter<ListItem>;
+
+    @Element()
+    public HostElement: HTMLLimelFileViewerElement;
+
+    private fullscreen: Fullscreen;
+
+    private fileType: FileType;
+
+    constructor() {
+        this.fullscreen = new Fullscreen(this.HostElement);
+        this.fileType = detectExtension(this.filename, this.url);
+    }
+
+    public render() {
+        this.sameDomain = this.url.includes(this.webserverDomain);
+
+        if (!this.url) {
+            return;
+        }
+
+        if (!this.isOfficeFilePubliclyAvailable) {
+            return this.renderNoFileSupportMessage();
+        }
+
+        switch (this.fileType) {
+            case 'pdf':
+                return this.renderPdf();
+            case 'image':
+                return this.renderImage();
+            case 'video':
+                return this.renderVideo();
+            case 'audio':
+                return this.renderAudio();
+            case 'text':
+                return this.renderText();
+            case 'office':
+                return this.renderOffice();
+            default:
+                return this.renderNoFileSupportMessage();
+        }
```

Comment:
I don't know why I am not a fan of switch. I will do it like this. And I will change the functions to arrow function
```suggestion
       return this.renderFileViewer();
       
       ...
           private renderFileViewer(){
        const fileViewerFunctions = {
            pdf: this.renderPdf,
            image: this.renderImage,
            video: this.renderVideo,
            audio: this.renderAudio,
            text: this.renderText,
            office: this.renderOffice,
        };
    
        const fileViewerFunction = fileViewerFunctions[this.fileType] || this.renderNoFileSupportMessage;
    
        return fileViewerFunction();
    }
       
```